docs/notes/feature-nothing.md

```diff
@@ -0,0 +1,5 @@
+# Nothing
+The constant `nothing` has been added to the language.
+
+The `nothing` constant resolved to the internal value kMCNull,
```

Comment:
The `nothing` constant resolves to a unique value (with unique type) which is used to represent the notion of no value.